server/src/utils/instance.ts

```diff
@@ -0,0 +1,79 @@
+// Instance tracking and management
+import { INSTANCE_TO_CLASS_MAP } from '../config/config';
+
+// Export the instance to class map from config
+export const instanceToClassMap = INSTANCE_TO_CLASS_MAP;
+
+export let instanceDefinitions: { [key: string]: string } = {};
```

Comment:
again an issue with my original implementation-- I don't think the instanceDefinitions are being reset between documents. potentially this could lead to stale data when a user switches files. maybe we should reset `instanceDefinitions` at the start of `trackInstanceDefinitions()` or per-document?